### PR TITLE
feat(mtp): add torch.compile and integration tests for MTP forward

### PR DESCRIPTION
**Pull request by rahul-tuli**
## Purpose

Make MTP forward `torch.compile`-compatible and add integration tests on par with Eagle3/DFlash/PEagle.

## Changes

**`src/speculators/models/mtp/core.py`**
- Add `@conditional_torch_compile` decorator to `forward()`
- Use uniform `valid_len = seq_len - num_steps - 1` across all loop iterations for stable compiled codegen
- Pre-compute causal mask once outside the loop (same shape every iteration)
- Use 3D `cross_entropy` via `logits.permute(0, 2, 1)` to avoid `reshape(-1, vocab_size)`

**`tests/integration/conftest.py`**
- Add `TINY_QWEN3_5_KWARGS` config dict and `make_mtp_model()` factory
- Add `preprocess` parameter to `make_batch()` for MTP's `shift_batch_mtp`

**`tests/integration/models/test_model_forward.py`**
- Add MTP to shared `ALL_SPECS` with `requires_transformers_version("5.2.0")` guard
- Add `hidden_size`, `hidden_multiplier`, `batch_factory` fields to `ModelSpec`
- Add `VOCAB_SPECS` (MTP excluded — vocab reduction not supported)
- Add `TestMTPParams` for varying `num_speculative_steps` [1, 2, 5]

**`tests/e2e/regression/test_mtp_online_acceptance.py`**
- Switch from Qwen3.5-9B to Qwen3.5-4B (fits 80GB GPUs)
- Adjust acceptance thresholds to match 4B base rates

## Tests

```bash
pytest tests/integration/models/test_model_forward.py -k mtp -v
pytest tests/integration/models/test_model_forward.py -v  # all models
```

## Checklist

- [x] Tests pass locally
- [x] `make style && make quality` pass
- [x] No new dependencies

### CODE REVIEW

File: examples/train/mtp_qwen3_5_9b_gsm8k_online.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 ### Example E2E run for Qwen3.5-9B on GSM8K ###
 
-# Timing (on 2x NVIDIA H100 80GB GPUs)
+# Timing (on 2x NVIDIA H200 141GB GPUs)
```
[the remark this answers is not in the document]
Comment (shanjiaz) [reply examples/train/mtp_qwen3_5_9b_gsm8k_online.sh:22]: @rahul-tuli could you update the readme too?

[the remark this answers is not in the document]
Comment (rahul-tuli) [reply examples/train/mtp_qwen3_5_9b_gsm8k_online.sh:22]: Removed this change from the current diff, since it really doesn't belong here, it's been moved to PR #596

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -48,6 +51,31 @@
     tie_word_embeddings=False,
 )
 
+TINY_QWEN3_5_KWARGS: dict = {
```
Comment (rahul-tuli) [inline tests/integration/conftest.py:54]: This is kept as a dict to avoid using too many if else checks, the associated config is loaded lazily for associated mtp tests which correctly skip when requirements are not met

**Comment by rahul-tuli** [comment]
e2e run: https://github.com/neuralmagic/llm-compressor-testing/actions/runs/27348290633

**Review by fynnsu — APPROVED** [review]
Looks good. Thanks for updating.
